fbw-common/src/wasm/systems/systems/src/apu/electronic_control_box.rs

```diff
@@ -175,20 +209,51 @@ impl ElectronicControlBox {
 
     pub fn update_fuel_pressure_switch_state(&mut self, fuel_pressure_switch: &FuelPressureSwitch) {
         if self.fault.is_none()
-            && 3. <= self.n.get::<percent>()
+            && 0. < self.n.get::<percent>()
             && !fuel_pressure_switch.has_pressure()
         {
             self.fault = Some(ApuFault::FuelLowPressure);
         }
     }
 
+    fn update_fuel_used(&mut self, context: &UpdateContext) {
+        self.fuel_used += self.fuel_flow
+            * MassConcentration::new::<kilogram_per_liter>(Self::JET_A_1_DENSITY)
+            * context.delta_as_time();
+    }
+
+    pub(super) fn update_fuel_used_reset(
+        &mut self,
+        engines: &[&impl EngineCorrectedN1],
+        lgciu: [&impl LgciuWeightOnWheels; 2],
+    ) {
+        let should_reset_fuel_used = (self.on_ground
+            && self.is_available()
+            && (self.engines_on
+                != engines
+                    .iter()
+                    .any(|&x| x.corrected_n1() > Ratio::new::<percent>(10.))))
+            || (!self.on_ground && lgciu.iter().all(|a| a.left_and_right_gear_compressed(true)))
+            || (!self.on_ground
+                && (self.turbine_state == TurbineState::Starting
+                    && (Duration::from_secs(2) == self.n_above_95_duration)));
```

Comment:
**question:** Why are we checking for exactly 2 seconds here? Shouldn't we check if the duration was above/below 2 seconds? It seems virtually impossible that the duration would be exactly 2 seconds on a single frame.

Reply:
The idea is to check for a condition that only happens once during start up, since this it to reset the fuel used. But you are absolutely right that this didn't work! I've changed it now to account for a range of N1 (and also added a test for it since I realised I forgot). This seems to work fine.